docs/src/tutorials/diffusion.jl

```diff
@@ -0,0 +1,273 @@
+# # A full example on how to use ClimaTimesteppers with ClimaCore
+#
+# In this tutorial, we will solve a diffusion equation on a sphere using
+# ClimaCore, and a mixed implicit-explicit solver in ClimaTimesteppers.
+#
+# First, we will set up the ClimaCore Spaces. We will define the implicit and
+# explicit tendencies and Jacobian. Next, we will set up the ODE problem and
+# solve it.
+#
+# ## The setup
+#
+# In this example, we consider a 3D spherical shell. We set initial data up on
+# the lower face of this shell, and we will see it diffuse horizontally and
+# vertically. We will treat the horizontal diffusion explicitly and the
+# vertical one implicitly.
+#
+# Let us start by importing the required pieces
+import SciMLBase
+import LinearAlgebra
+import ClimaTimeSteppers
+import ClimaCore
+import Plots
+import ClimaCore.MatrixFields: @name, ⋅
+
+# We also define some units, mostly to make the code more explicit
+const meters = meter = 1.0
+const kilometers = kilometer = 1.0
+const seconds = second = 1.0
```

Comment:
This is essentially documentation. I think it very useful to have precisely because this is a tutorial. But I realized I kilometers should be 1000 meters.

Units are never clear and people who are not familiar with a given code always have to understand what units are being used. This simple snippet dispels confusion and leads to a more clear `radius = 6700kilometers` instead of the more opaque `radius = 6.7e6`.

Reply:
Ok! This isn't something I've seen us do before, so we might want to keep things more similar to our actual driver code. But setting a kilometer to 1000 meters will make this less confusing, so I'm okay with that.